readinglist/authentication.py

```diff
@@ -18,7 +20,15 @@ def check_credentials(username, password, request):
     """
     settings = request.registry.settings
     is_enabled = settings.get('readinglist.basic_auth_backdoor')
-    return ["basicauth_%s" % username] if is_enabled else []
+
+    hmac_key = request.registry.settings.get(
```

Comment:
Also, we probably want to read this from the `__init__` and pass it as an argument rather than reading it from the authentication module.